synchronizer/l1_rollup_info_producer.go

```diff
@@ -164,13 +164,14 @@ func newL1DataRetriever(cfg configProducer, ethermans []EthermanInterface, outgo
 // ResetAndStop: reset the object and stop the current process. Set first block to be retrieved
 func (l *l1RollupInfoProducer) ResetAndStop(startingBlockNumber uint64) {
 	log.Infof("producer: Reset L1 sync process to blockNumber %d st=%s", startingBlockNumber, l.toStringBrief())
-	l.mutex.Lock()
-	defer l.mutex.Unlock()
 	log.Debugf("producer: Reset(%d): stop previous run (state=%s)", startingBlockNumber, l.status.String())
-	l.stopUnsafe()
+	l.Stop()
```

Comment:
Is the stop called from somewhere without needing the lock?? Could we just have the stop() function, instead the stop() and stopUnsafe()???

Reply:
Yes, the function stopUnsafe is already deleted!